### PR TITLE
Added support for TeamCity Role Assignments for User Groups

### PR DESCRIPTION
**Pull request by eli-guidewire**
* created a TeamCity resource (resource_group_role_assignment) which can assign to groups and unassign roles from groups
* added tests for the group role assignment resource
* updated the TeamCity main-config.xml to enable the per project permissions

This pull request relies on the pull request from the go-teamcity project:
https://github.com/cvbarros/go-teamcity/pull/86

### CODE REVIEW

**Comment by eli-guidewire** [comment]
Note that CI tests are going to fail until the reference to go-teamcity is updated.

**Comment by sundarok** [comment]
Hi @cvbarros Can you please review and merge this PR?

**Comment by cvbarros** [comment]
Hi @eli-guidewire , would you be so kind to please rebase onto current master after #89 is merged. That way we can have the CI checks - once done we'll get this merged and released asap.
Thanks!

File: integration_tests/start_teamcity.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
 set -e
 
 pushd integration_tests/
-
-bsdtar -xzf teamcity_data.tar.gz
+# using tar which is readily available whereas bsdtar is not even available on some linux distros
+tar -xzf teamcity_data.tar.gz
```
Comment (cvbarros) [inline integration_tests/start_teamcity.sh:6]: As commented here: https://github.com/cvbarros/terraform-provider-teamcity/pull/88#discussion_r439060316

**Comment by justinto-guidewire** [comment]
Hi @cvbarros, sorry we missed your latest comment in the PR, where would you like us to add documentation for this new resource?

**Comment by justinto-guidewire** [comment]
We'll have to do some clean up on this branch first and maybe create another PR if necessary

**Comment by justinto-guidewire** [comment]
I've forked another repo and recreated this PR, see: https://github.com/cvbarros/terraform-provider-teamcity/pull/90

Feel free to close this PR as I don't see an option for me to close myself...

**Pull request closed without merging**